sdk/go/things_test.go

```diff
@@ -0,0 +1,588 @@
+		err   error
+	}{
+
+		{
+			desc:  "connect existing thing to existing channel",
+			thId:  "1",
+			chId:  "1",
+			token: token,
+			err:   nil,
+		},
+
+		{
+			desc:  "connect existing thing to non-existing channel",
+			thId:  "1",
+			chId:  "9",
+			token: token,
+			err:   sdk.ErrNotFound,
+		},
+		{
+			desc:  "connect non-existing thing to existing channel",
+			thId:  "9",
+			chId:  "1",
+			token: token,
+			err:   sdk.ErrNotFound,
+		},
+		{
+			desc:  "connect existing thing to channel with invalid ID",
+			thId:  "1",
+			chId:  "invalid",
+			token: token,
+			err:   sdk.ErrFailedConnection,
+		},
+		{
+			desc:  "connect thing with invalid ID to existing channel",
+			thId:  "invalid",
+			chId:  "1",
+			token: token,
+			err:   sdk.ErrFailedConnection,
+		},
+
+		{
+			desc:  "connect existing thing to existing channel with invalid token",
+			thId:  "1",
+			chId:  "1",
+			token: wrongValue,
+			err:   sdk.ErrUnauthorized,
+		},
+		{
+			desc:  "connect existing thing to existing channel with empty token",
+			thId:  "1",
+			chId:  "1",
+			token: "",
+			err:   sdk.ErrUnauthorized,
+		},
+		{
+			desc:  "connect thing from owner to channel of other user",
+			thId:  "1",
+			chId:  "2",
+			token: token,
+			err:   sdk.ErrNotFound,
+		},
+	}
+
+	for _, tc := range cases {
+		err := mainfluxSDK.ConnectThing(tc.thId, tc.chId, tc.token)
+		assert.Equal(t, tc.err, err, fmt.Sprintf("%s: expected error %s, got %s", tc.desc, tc.err, err))
+	}
+}
+
+func TestDisconnectThing(t *testing.T) {
+	svc := newThingsService(map[string]string{
+		token:      email,
+		otherToken: otherEmail,
+	})
+
+	ts := newThingsServer(svc)
+	defer ts.Close()
+	sdkConf := sdk.Config{
+		BaseURL:           ts.URL,
+		UsersPrefix:       "",
+		ThingsPrefix:      "",
+		HTTPAdapterPrefix: "",
+		MsgContentType:    contentType,
+		TLSVerification:   false,
+	}
+
+	mainfluxSDK := sdk.NewSDK(sdkConf)
+	mainfluxSDK.CreateThing(thing, token)
+	mainfluxSDK.CreateChannel(channel, token)
+	mainfluxSDK.ConnectThing("1", "1", token)
+	mainfluxSDK.CreateChannel(channel, otherToken)
+
+	cases := []struct {
+		desc  string
+		thId  string
+		chId  string
+		token string
+		err   error
+	}{
+
+		{
+			desc:  "disconnect connected thing from channel",
+			thId:  "1",
+			chId:  "1",
+			token: token,
+			err:   nil,
+		},
+
+		{
+			desc:  "disconnect existing thing from non-existing channel",
+			thId:  "1",
+			chId:  "9",
+			token: token,
+			err:   sdk.ErrNotFound,
+		},
+		{
+			desc:  "disconnect non-existing thing from existing channel",
+			thId:  "9",
+			chId:  "1",
+			token: token,
+			err:   sdk.ErrNotFound,
+		},
+		{
+			desc:  "disconnect existing thing from channel with invalid ID",
+			thId:  "1",
+			chId:  "invalid",
+			token: token,
+			err:   sdk.ErrFailedConnection,
+		},
+		{
+			desc:  "disconnect thing with invalid ID from existing channel",
+			thId:  "invalid",
+			chId:  "1",
+			token: token,
+			err:   sdk.ErrFailedConnection,
```

Comment:
It's because of error in https://github.com/mainflux/mainflux/blob/master/sdk/go/sdk.go#L46
Definitions of sdk.ErrFailedConnection and sdk.ErrFailedDisconnect was the same there, it's fixed now